### PR TITLE
feat: #4407 v3 component refactor

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #4407 

Add floating button and button group components.

### CODE REVIEW

File: packages/elements/src/components/button/__styles__/index.tsx [hunk 1]
```diff
@@ -113,13 +141,52 @@ export const ElButton = styled.button`
...
     }
   }
+
+  ${isMobile} {
+    height: 2.5rem;
+    font-size: 1rem;
+    padding: ${buttonYPaddingMobile}rem 0.75rem;
+  }
+
+  &.el-floating-button {
```
Comment (willmcvay) [inline packages/elements/src/components/button/__styles__/index.tsx:151]: `.${elFloatingButton} {`

Comment (willmcvay) [reply packages/elements/src/components/button/__styles__/index.tsx:151]: If the consuming app hashes the class name, `el-floating-button` won't be a thing!

File: packages/elements/src/components/button/__styles__/index.tsx [hunk 2]
```diff
@@ -113,13 +141,52 @@ export const ElButton = styled.button`
...
+    justify-content: center;
+    align-items: center;
+
+    ${isMobile} {
+      width: 2.5rem;
+      height: 2.5rem;
+      margin: 0.5rem;
+    }
+
+    &${elIsLoading} {
```
Comment (willmcvay) [inline packages/elements/src/components/button/__styles__/index.tsx:168]: Needs a `.` ie `&.${elIsLoading}`

File: packages/elements/src/components/button/__styles__/index.tsx [hunk 3]
```diff
@@ -113,13 +141,52 @@ export const ElButton = styled.button`
...
+      &:after {
+        left: inherit;
+        top: inherit;
+      }
+
+      .el-icon {
```
Comment (willmcvay) [inline packages/elements/src/components/button/__styles__/index.tsx:174]: As above, should be `.${elIcon}`

File: packages/elements/src/components/button/__styles__/index.tsx [hunk 4]
```diff
@@ -217,3 +288,36 @@ export const elButtonHasRightChevron = css`
...
   }
 `
+
+export const ElButtonGroup = styled.div`
+  display: grid;
+  grid-auto-flow: column;
+  column-gap: 1rem;
+  row-gap: 1rem;
+
+  .el-button-size2 {
```
Comment (willmcvay) [inline packages/elements/src/components/button/__styles__/index.tsx:298]: Same as above, the three button classes should be the raw class name

File: packages/elements/src/components/button/index.tsx [hunk 5]
```diff
@@ -13,6 +14,18 @@ export interface IButton extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   chevronRight?: boolean
   fullWidth?: boolean
   className?: string
+  size?: 2 | 3 | 4
```
Comment (willmcvay) [inline packages/elements/src/components/button/index.tsx:17]: Should size be a type?

File: packages/elements/src/components/button/index.tsx [hunk 6]
```diff
@@ -13,6 +14,18 @@ export interface IButton extends React.ButtonHTMLAttributes<HTMLButtonElement> {
...
+}
+
+const resolveButtonSize = (size: 2 | 3 | 4): string => {
```
Comment (willmcvay) [inline packages/elements/src/components/button/index.tsx:20]: Could use a test :-)

**Review by willmcvay — CHANGES_REQUESTED** [review]

File: packages/elements/src/components/button/__styles__/index.tsx (hunk 2 above)
Comment (bashleigh) [reply packages/elements/src/components/button/__styles__/index.tsx:169]: I don't quite get this one. Sometimes I add a dot sometimes I don't 🤷‍♀️   in this case, if I add the dot, the applied styles disappear 🤷‍♀️

**Review by willmcvay — APPROVED** [review]

**Pull request merged**